### DISCUSSION TITLE
GCP Key restrictions

### DISCUSSION THREAD

**Discussion by votemike** [vercel/community#2828] [Help] [answered]
### Summary

I am trying to restrict the Places API key in my GCP console, but cannot get it working with the Vercel deploy URLs.
I believe the most appropriate restriction is a Website Restriction: https://cloud.google.com/docs/authentication/api-keys?_ga=2.155461112.-2123857779.1682315182#adding_http_restrictions

I am trying a simple use-case of `*.vercel.app/*` (I know that this would allow anyone deploying to Vercel to use the key), but even this broad restriction seems to block the use of my key on the Vercel preview URLs.

Any suggestions on what I'm doing wrong or how to get around this?

Thanks

### Steps to Reproduce

Restrict a key in GCP to specific websites (e.g. `*.vercel.app/*`)

Try to use that key in a project deployed to a Vercel preview URL https://my-project.vercel.app/something

The response when using the key is a 403.

**Comment by smaeda-ks** [comment, marked as answer]
@votemike 
Places API is under the "Web Service APIs" category:
https://developers.google.com/maps/apis-by-platform#web_service_apis

and they don't support the HTTP referer restriction for those APIs:
https://developers.google.com/maps/faq#browser-keys-blocked-error

> You are using any of the [web service APIs](https://developers.google.com/maps/apis-by-platform#web_service_apis) with an API key restricted to an HTTP referer. For security reasons, web service APIs need to use API keys restricted to IP addresses. [Switch your key restriction type from an HTTP referer restriction to an IP address restriction](https://developers.google.com/maps/faq#switch-key-type), or create a new API key if your key is already used with the Maps JavaScript API.

Unfortunately, our serverless/edge functions use [dynamic IPs](https://vercel.com/guides/can-i-get-a-fixed-ip-address). Only Enterprise plan can have a fixed outbound IP addresses:
https://vercel.com/docs/security/secure-compute
